src/cmds/qstat.c

```diff
@@ -1329,17 +1330,17 @@ display_statjob(struct batch_status *status, struct batch_status *prtheader, int
 					}
 				}
 				a = a->next;
-				if ((a || output_format == FORMAT_DEFAULT))
+				if (a)
 					printf("%s", delimiter);
-				else if (output_format == FORMAT_JSON) {
-					if (prev_resc_name != NULL)
-						if (add_json_node(JSON_OBJECT_END, JSON_NULL, JSON_NOVALUE, NULL,
-								NULL) == NULL)
-							return 1;
-					if (add_json_node(JSON_OBJECT_END, JSON_NULL, JSON_NOVALUE, NULL, NULL)
-							== NULL)
+			}
+			if (output_format == FORMAT_DEFAULT)
+					printf("%s", delimiter);
+			else if (output_format == FORMAT_JSON) {
+				if (prev_resc_name != NULL)
+					if (add_json_node(JSON_OBJECT_END, JSON_NULL, JSON_NOVALUE, NULL, NULL) == NULL)
 						return 1;
-				}
+				if (add_json_node(JSON_OBJECT_END, JSON_NULL, JSON_NOVALUE, NULL, NULL) == NULL)
```

Comment:
Attributes could be represented 3 different ways in json: string, number, object.
If the attribute is of type resource list, the node has a key of the attribute name, the value is a new object with keys resource names and values resource values.

Let's take this example of a job:
```JSON
"15.shecil":{
    "Resource_List":{
        "ncpus":1,
        "nodect":1,
        "place":"pack",
        "select":"1:ncpus=1"
    }
}
```
`Resource_List` starts a new object, but it has to be ended before we enter the next job, otherwise the next job would be under this job.

Reply:
Ok, thanks for clarifying